HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,244 @@
+        std::cout << transformationMatrix.matrix() << std::endl;
+        fileStorageIn.release();
+
+        // Extract Rotation Matrix and Translation Vector from Transformation Matrix
+        std::cout << "RotationMatrix:\n" << transformationMatrix.linear() << std::endl;
+        std::cout << "TranslationVector:\n" << transformationMatrix.translation() << std::endl;
+
+        // Transform from Zivid to Robot (Rotation Matrix --> any format)
+        const auto robotRepresentations = transformFromZividToRobot(transformationMatrix);
+
+        // Transform from Robot to Zivid (any format --> Rotation Matrix)
+        const auto transformationMatrixFromQuaternion =
+            transformFromRobotToZivid(transformationMatrix, robotRepresentations);
+        // Save Quaternion --> Transformation Matrix to .YAML file
+        auto fileStorageOut = cv::FileStorage{};
+        if(!fileStorageOut.open("robotTransformOut.yaml", cv::FileStorage::Mode::WRITE))
+        {
+            throw std::runtime_error("Could not open robotTransformOut.yaml for writing");
+        }
+        fileStorageOut.write("TransformationMatrixFromQuaternion",
+                             eigenToCv(transformationMatrixFromQuaternion.matrix()));
+        fileStorageOut.release();
+    }
+
+    catch(const std::exception &e)
+    {
+        std::cerr << "Error: " << e.what() << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+cv::Mat eigenToCv(const Eigen::MatrixXd &eigenMat)
+{
+    cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+    cv::eigen2cv(eigenMat, cvMat);
+
+    return cvMat;
+}
+
+Eigen::MatrixXd cvToEigen(const cv::Mat &cvMat)
+{
+    Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+    cv::cv2eigen(cvMat, eigenMat);
+
+    return eigenMat;
+}
+
+Representations transformFromZividToRobot(const Eigen::Affine3d &transformationMatrix)
+{
+    Representations robotRepresentations;
+    std::cout << "\nConverting Rotation Matrix to Axis-Angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle(transformationMatrix.linear());
+    std::cout << "Axis:\n" << axisAngle.axis() << std::endl;
+    std::cout << "Angle:\n" << axisAngle.angle() << std::endl;
+
+    // Axis-angle to Rotation Vector
+    std::cout << "\nConverting Axis-Angle to Rotation Vector:" << std::endl;
+    if(!axisAngle.axis().isUnitary())
+    {
+        std::cout << "Note! Axis in Axis-Angle pair is not unitary" << std::endl;
+    }
+    robotRepresentations.rotationVector = axisAngle.angle() * axisAngle.axis();
+    std::cout << robotRepresentations.rotationVector << std::endl;
+
+    // Rotation Matrix to Quaternion
+    std::cout << "\nConverting Rotation Matrix to Quaternion:" << std::endl;
+    const Eigen::Quaterniond quaternion(transformationMatrix.linear());
+    robotRepresentations.quaternion = quaternion;
+    std::cout << robotRepresentations.quaternion.coeffs() << std::endl;
```

Comment:
We don't comment on whether this is qw qx qy qz or qx qy qz qw. Is that clear from Eigen or should we add a line about that?